kolibri/core/content/api.py

```diff
@@ -1321,6 +1326,78 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentRequestViewset(ReadOnlyValuesViewset, CreateModelMixin):
+    serializer_class = serializers.ContentDownloadRequestSeralizer
+
+    pagination_class = OptionalPageNumberPagination
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+        "facility",
+        "source_id",
+    )
+
+    def get_queryset(self):
+        return ContentDownloadRequest.objects.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return queryset.annotate(
+            has_removal=Exists(
+                ContentRemovalRequest.objects.filter(
+                    source_model=OuterRef("source_model"),
+                    source_id=OuterRef("source_id"),
+                    contentnode_id=OuterRef("contentnode_id"),
+                    requested_at__gte=OuterRef("requested_at"),
+                ).exclude(status=ContentRequestStatus.Failed)
+            )
+        ).filter(has_removal=False)
+
+    def delete(self, request, pk=None):
+        request_id = pk
+
+        if request_id is None:
+            return Response(
+                {"detail": "Request ID is required"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        existing_download_request = (
+            existing_deletion_request
+        ) = ContentRemovalRequest.objects.filter(
+            id=request_id,
+            source_id=request.user.id,
+        ).first()
+
+        if existing_download_request is None:
+            return Response(
+                {"detail": "No existing download request found"},
+                status=status.HTTP_204_NO_CONTENT,
+            )
+
+        existing_deletion_request = ContentRemovalRequest.objects.filter(
+            id=request_id,
+            source_id=request.user.id,
+        ).first()
+
+        if existing_deletion_request:
+            if existing_deletion_request.status == ContentRequestStatus.Failed:
+                existing_deletion_request.status = ContentRequestStatus.Pending
+                existing_deletion_request.save()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        content_request = ContentRemovalRequest.build_for_user(request.user)
+        content_request.contentnode_id = existing_download_request.contentnode_id
+        content_request.save()
+
+        automatic_resource_import.enqueue()
```

Comment:
Very small thing, seems the task enqueue is skipped if there was an existing deletion. So I think:
1) remove the `return` on line 1391
2) add an `else` for `if existing_deletion_request:` and put lines 1393-1395 inside

Reply:
Seems like calling this on every invocation might lead to a lot of queued tasks? Should we be doing something more like 'enqueue if not running'?

Reply:
we definitely want it enqueued whenever something changes, so we do need it when we reset the status on the existing request. I think an optimal approach would be 'enqueue if not already queued', but if one is still running, I think it's wise to just enqueue another task to avoid race conditions leading to unprocessed downloads.

Reply:
So there isn't a utility within the job mechanism to do that, right? With the exception of using a static job ID

Reply:
No - I can file a follow up issue for this so as not to block this PR.

Reply:
Issue here: https://github.com/learningequality/kolibri/issues/11007 I self assigned.